src/browser-polyfill.js

```diff
@@ -132,7 +132,39 @@ if (typeof browser === "undefined") {
         }
 
         return new Promise((resolve, reject) => {
-          target[name](...args, makeCallback({resolve, reject}, metadata));
+          function callAPIWithNoCallback() {
+            try {
+              target[name](...args);
+
+              // Update the API method metadata, so that the next API calls will not try to
+              // use the unsupported callback anymore.
+              metadata.fallbackToNoCallback = false;
+              metadata.noCallback = true;
+
+              resolve();
+            } catch (error) {
+              // Catch API parameters validation errors and rejects the promise.
+              reject(error);
+            }
+          }
```

Comment:
Now the `callAPIWithNoCallback` logic is very simple, it can be inlined below, without try-catch (because you are in the closure of a Promise).

The second `callAPIWithNoCallback` call does not require the metadata to be updated, so it can simply be:

```javascript
target[name](...args);
resolve();
```